core/src/test/java/org/elasticsearch/test/ESBlobStoreTestCase.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.test;
+
+import org.elasticsearch.common.blobstore.BlobContainer;
+import org.elasticsearch.common.blobstore.BlobPath;
+import org.elasticsearch.common.blobstore.BlobStore;
+import org.elasticsearch.common.bytes.BytesArray;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+/**
+ * Generic test case for blob store implementation.
+ * These tests check basic blob store functionality.
+ */
+public abstract class ESBlobStoreTestCase extends ESTestCase {
+
+    public void testContainerCreationAndDeletion() throws IOException {
+        final BlobStore store = newBlobStore();
+        final BlobContainer containerFoo = store.blobContainer(new BlobPath().add("foo"));
+        final BlobContainer containerBar = store.blobContainer(new BlobPath().add("bar"));
+        byte[] data1 = randomBytes(randomIntBetween(10, scaledRandomIntBetween(1024, 1 << 16)));
+        byte[] data2 = randomBytes(randomIntBetween(10, scaledRandomIntBetween(1024, 1 << 16)));
+        containerFoo.writeBlob("test", new BytesArray(data1));
+        containerBar.writeBlob("test", new BytesArray(data2));
+
+        assertArrayEquals(readBlobFully(containerFoo, "test", data1.length), data1);
+        assertArrayEquals(readBlobFully(containerBar, "test", data2.length), data2);
+
+        assertTrue(containerFoo.blobExists("test"));
+        assertTrue(containerBar.blobExists("test"));
+        store.delete(new BlobPath());
+        assertFalse(containerFoo.blobExists("test"));
+        assertFalse(containerBar.blobExists("test"));
+
+        store.close();
+    }
+
+    public static byte[] createRandomBlob(BlobContainer container, String name, int length) throws IOException {
```

Comment:
I'd rename this method to `writeRandomBlob()` since this is generating random data and writes it to the container